### PR TITLE
Generate AOI

### PR DESCRIPTION
**Pull request by MohamedHamouGisaia**
image

image

### CODE REVIEW

File: projects/arlas-components/assets/i18n/fr.json [hunk 1]
```diff
@@ -75,5 +75,14 @@
...
 	"Fit the thumbnail's width to the tile": "Ajuster la largeur de la vignette à la tuile",
-	"Fit the thumbnail's height to the tile": "Ajuster la hauteur de la vignette à la tuile"
+	"Fit the thumbnail's height to the tile": "Ajuster la hauteur de la vignette à la tuile",
+	"Enter coordinates in decimal or sexagesimal degrees": "Saisir les coordonnées en degrés décimaux ou sexagésimaux.",
+	"BBox generator": "Générer une bbox",
+	"First corner": "Premier coin",
+	"Second corner": "Deuxième coin",
+	"You must enter a coordinate": "Saisir une coordonnée",
+	"Enter a coordinate in decimal (1.1) or sexagesimal (1° 6' 3\")": "Saisir une coordonnée en décimal (1.1) ou en sexagésimal (1° 6' 3\")",
+	"Decimal: 1.1 or Sexagesimal 1°6'3\" coordinate": "Coordonnée en décimal: 1.1 or sexagésimal: 1°6'3\"",
```
Comment (QuCMGisaia) [inline projects/arlas-components/assets/i18n/fr.json:85]: or -> ou en

File: projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.scss [hunk 2]
```diff
@@ -0,0 +1,68 @@
+.wrapper {
+    padding: 5px;
+    border-radius: 3px;
```
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.scss:3]: Should be 4px to be uniform with other WUI

File: projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    private cdr: ChangeDetectorRef,
+    @Inject(MAT_DIALOG_DATA) public data: {
+      initCorner: Corner;
+    },
+    public dialogRef: MatDialogRef<BboxGeneratorComponent>,) {
+  }
+
+  public ngOnInit(): void {
+    if (!!this.data && !this.data.initCorner) {
+      this.data.initCorner = {
```
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts:47]: Is there a reason for this coordinates ? Could it be linked to the current extent of the map ?

File: projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    }
+    this.bboxForm = new BboxFormGroup(this.data.initCorner);
+  }
+  public ngAfterViewInit(): void {
+    this.cdr.detectChanges();
+  }
+
+  public getErrorMessage(formControl: FormControl | FormGroup) {
+    if (formControl.hasError('required')) {
+      return 'You must enter a coordinate';
```
Comment (QuCMGisaia) [inline projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts:60]: If you use a marker on those texts, we can add them to the translated keys more easily

File: projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    },
+    public dialogRef: MatDialogRef<BboxGeneratorComponent>,) {
+  }
+
+  public ngOnInit(): void {
+    if (!!this.data && !this.data.initCorner) {
+      this.data.initCorner = {
+        lat: 32,
+        lng: -8
+      };
```
Comment (sebbousquet) [inline projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts:50]: Why this default values ?

File: projects/arlas-components/assets/i18n/fr.json [hunk 6]
```diff
@@ -75,5 +75,14 @@
...
+	"Both corners have the same latitudes, modify one of them.": "Les deux coins ont les mêmes latitudes, modifiez l'un d'entre eux.",
```
Comment (sebbousquet) [inline projects/arlas-components/assets/i18n/fr.json:86]: "Les deux coins ont la même latitude"

File: projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts (hunk 3 above)
Comment (MohamedHamouGisaia) [reply projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts:47]: the right value will be wired on the wui side

File: projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts (hunk 4 above)
Comment (MohamedHamouGisaia) [reply projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts:60]: i'll do that in another pr

File: projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts (hunk 5 above)
Comment (MohamedHamouGisaia) [reply projects/arlas-components/src/lib/components/bbox-generator/bbox-generator.component.ts]: I changed it to 0 0

**Pull request merged**